.github/workflows/spellcheck.yml

```diff
@@ -0,0 +1,44 @@
+# Copyright (C) 2022 Daniel Stenberg, <daniel@haxx.se>, et al.
+#
+# SPDX-License-Identifier: curl
+
+name: spell
+on:
+  push:
+    branches:
+    - master
+    paths:
+    - '**.md'
+    - '!docs/BINDINGS.md'
+  pull_request:
+    branches:
+    - master
```

Comment:
I guess the same filter should then also apply to pull_requests and both filters also need to include the relevant files under `.github/workflows`